lib/palindrome_check.rb

```diff
@@ -1,5 +1,31 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+
+  # delete all whitespace
+  # return false if it's not a string
+  unless my_phrase.is_a? String
+    return false
+  end
+
+stripped_phrase = ""
```

Comment:
Creating a new string, i.e. `stripped_phrase` will increase the time complexity. Can you think of an algorithm that does not need this additional copy of the string and can skip white spaces as it examines them?